plugins/dev-create/bin/app.js

```diff
@@ -0,0 +1,64 @@
+const chalk = require('chalk');
+const fs = require('fs-extra');
+const path = require('path');
+const execSync = require('child_process').execSync;
+const {createDir} = require('./common');
+
+exports.createApp = function(name, options) {
+  if (options.typescript) {
+    console.error('Not yet implemented');
+    process.exit(1);
+  }
+
+  const root = process.cwd();
+  const dir = options.dir ?? name;
+  const appPath = path.join(root, dir);
+
+  // Create the new directory for the app if needed.
+  createDir(appPath);
+  console.log(
+      `Creating a new Blockly application called ${name} in ${appPath}`);
+
+  // Copy over files from sample-app directory.
+  const sampleDir = '../../../examples/sample-app';
+  fs.copySync(path.resolve(__dirname, sampleDir), appPath, {
+    filter(src) {
+      const excludes = [
+        'node_modules', 'dist', 'package-lock.json', 'package.json'];
+      for (const file of excludes) {
+        if (src.indexOf(file) > -1) return false;
+      }
+      return true;
+    },
+  });
+
+  // Create package.json file for app.
+  const templateJson = require(path.join(sampleDir, 'package.json'));
```

Comment:
Any particular reason you opt to read this from the filesystem whereas for plugins it's hardcoded into the script?

Reply:
The sample app directory is a standalone directory we are copying and I hope to put that on the examples site soon, so it needs a real package.json. It wouldn't make sense to also have this information hardcoded into the script, because you'd have to update it in both places.

The plugin template used to have both a template file and some hardcoded sections, but the template files were not useful because there was one for each type of plugin, and all of them were entirely redundant with the hardcoded version. Those templates are not standalone packages so they didn't need the template file, and the plugin script does a lot more modification of the values present anyway, so it makes more sense for that to be generated entirely within the script.

Reply:
Sounds sensible to me!